codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/InterceptorConfigCustomization.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.client.smithy.customizations
+
+import software.amazon.smithy.rust.codegen.client.smithy.generators.config.ConfigCustomization
+import software.amazon.smithy.rust.codegen.client.smithy.generators.config.ServiceConfig
+import software.amazon.smithy.rust.codegen.core.rustlang.rust
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.rustlang.writable
+import software.amazon.smithy.rust.codegen.core.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+
+class InterceptorConfigCustomization(codegenContext: CodegenContext) : ConfigCustomization() {
+    private val moduleUseName = codegenContext.moduleUseName()
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val interceptors = RuntimeType.smithyRuntimeApi(runtimeConfig).resolve("client::interceptors")
+    private val codegenScope = arrayOf(
+        "Interceptor" to interceptors.resolve("Interceptor"),
+        "SharedInterceptor" to interceptors.resolve("SharedInterceptor"),
+    )
+
+    override fun section(section: ServiceConfig) =
+        writable {
+            when (section) {
+                ServiceConfig.ConfigStruct -> rustTemplate(
+                    """
+                    // TODO(enableNewSmithyRuntime): Unused until we completely switch to the orchestrator
+                    ##[allow(dead_code)]
```

Comment:
This customization shouldn't run unless we're generating the orchestrator. Otherwise, this will pull in the runtime crates.

It looks like this is done correctly where the customization is registered, so this `allow` shouldn't be necessary?

Reply:
Thanks for catching! Addressed in 73c3515.